cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator.cpp

```diff
@@ -0,0 +1,472 @@
+ * \note
+ *   ROS package name: cob_twist_controller
+ *
+ * \author
+ *   Author: Christoph Mark, email: christoph.mark@ipa.fraunhofer.de / christoph.mark@gmail.com
+ *
+ * \date Date of creation: June, 2015
+ *
+ * \brief
+ *   This module contains the implementation of all interface types.
+ *
+ ****************************************************************/
+#define RAMP_PROFILE 1
+#define SINOID_PROFILE 2
+#include "cob_cartesian_controller/trajectory_profile_generator/trajactory_profile_generator.h"
+
+/* BEGIN TrajectoryProfileBase *****************************************************************************************/
+
+TrajectoryProfileBase* TrajectoryProfileBuilder::createProfile(const cob_cartesian_controller::CartesianActionStruct& params)
+{
+    TrajectoryProfileBase* ib = NULL;
+    switch(params.profile.profile_type)
+    {
+        case RAMP_PROFILE:
+            ib = new TrajectoryProfileRamp(params);
+            break;
+        case SINOID_PROFILE:
+            ib = new TrajectoryProfileSinoid(params);
+            break;
+        default:
+            ROS_ERROR("Unknown Profile");
+            break;
+    }
+
+    return ib;
+}
+/* END TrajectoryProfileBase *******************************************************************************************/
+
+/* BEGIN TrajectoryProfileRamp ********************************************************************************************/
+inline void TrajectoryProfileRamp::getProfileTimings(double Se_max, double accl, double vel)
+{
+        int steps_te, steps_tv, steps_tb = 0;
+        double tv, tb, te = 0.0;
+
+        // Calculate the Ramp Profile Parameters
+        if (vel > sqrt(std::fabs(Se_max) * accl))
+        {
+            vel = sqrt(std::fabs(Se_max) * accl);
+        }
+        tb = vel / accl;
+        te = (std::fabs(Se_max) / vel) + tb;
+        tv = te - tb;
+
+        // Interpolationsteps for every timesequence
+        steps_tb = trunc(tb      / params_.profile.t_ipo);
+        steps_tv = trunc((tv-tb) / params_.profile.t_ipo);
+        steps_te = trunc((te-tv) / params_.profile.t_ipo);
+
+        // Reconfigure timings wtih t_ipo_
+        pt_.tb = steps_tb * params_.profile.t_ipo;
+        pt_.tv = (steps_tb + steps_tv) * params_.profile.t_ipo;
+        pt_.te = (steps_tb + steps_tv + steps_te) * params_.profile.t_ipo;
+        pt_.steps_tb = steps_tb;
+        pt_.steps_tv = steps_tv;
+        pt_.steps_te = steps_te;
+}
+
+inline bool TrajectoryProfileRamp::generatePath(cob_cartesian_controller::PathArray &pa)
+{
+    std::vector<double> array;
+    double accl = params_.profile.accl;
+
+    double direction = pa.getSe()/std::fabs(pa.getSe());
+
+    if(pa.getCalcTe())
+    {
+        ROS_INFO_STREAM("Longest Se.. Vel_max: " << params_.profile.vel << "   accl_max: " << params_.profile.accl << "   tb: " << pt_.tb << "   tv: " << pt_.tv << "   te: " << pt_.te);
+
+        int i = 1;
+        // Calculate the ramp profile path
```

Comment:
The following lines look pretty much the same as in 162....I bet this can be unified somehow!